src/pkcs15init/profile.c

```diff
@@ -1247,7 +1247,8 @@ new_file(struct state *cur, const char *name, unsigned int type)
 	} else if (!strcasecmp(name+7, "AppDF")) {
 		file = init_file(SC_FILE_TYPE_DF);
 	} else {
-		if (map_str2int(cur, name+7, &df_type, pkcs15DfNames))
+		if (map_str2int(cur, name+7, &df_type, pkcs15DfNames)
+				|| df_type >= SC_PKCS15_DF_TYPE_COUNT)
```

Comment:
well, as promised, formatting will be an endless spring of discussion 😉 
I personally like
- short lines, which I think are more readable
- having the logical operator in front, because with many conditions, you'll have it always in the same column and it will always be two characters long (i.e. `||` or `&&`)

But of course I admit that the logical operators in front is not used very often and I'm not sure if clang-format even offers an option for this.

I'm OK with simply changing this to the style that is currently proposed with https://github.com/OpenSC/OpenSC/pull/2017)